### PR TITLE
Improve decoding of constant composites

### PR DESCRIPTION
**Pull request by AlexeySotkin**
Signed-off-by: Alexey Sotkin <alexey.sotkin@intel.com>

### CODE REVIEW

File: lib/SPIRV/libSPIRV/SPIRVStream.cpp [hunk 1]
```diff
@@ -296,4 +296,29 @@ spv_ostream &operator<<(spv_ostream &O, const SPIRVNL &E) {
   return O;
 }
 
+// Read the next word from the stream and if OpCode matches the argument,
+// decode the whole instruction. Multiple such instruction are possible.
```
Comment (Fznamznon) [inline lib/SPIRV/libSPIRV/SPIRVStream.cpp:300]:
```suggestion
// decode the whole instruction. Multiple such instructions are possible.
```

File: lib/SPIRV/libSPIRV/SPIRVStream.cpp [hunk 2]
```diff
@@ -296,4 +296,29 @@ spv_ostream &operator<<(spv_ostream &O, const SPIRVNL &E) {
...
+// If OpCode doesn't match the arguemnt, set position of the next character to
```
Comment (Fznamznon) [inline lib/SPIRV/libSPIRV/SPIRVStream.cpp:301]:
```suggestion
// If OpCode doesn't match the argument, set position of the next character to
```

File: lib/SPIRV/libSPIRV/SPIRVStream.cpp [hunk 3]
```diff
@@ -296,4 +296,29 @@ spv_ostream &operator<<(spv_ostream &O, const SPIRVNL &E) {
...
+// be extracted from the stream to the beginig of the non-matching instruction.
```
Comment (Fznamznon) [inline lib/SPIRV/libSPIRV/SPIRVStream.cpp:302]:
```suggestion
// be extracted from the stream to the beginning of the non-matching instruction.
```

File: lib/SPIRV/libSPIRV/SPIRVStream.cpp [hunk 4]
```diff
@@ -296,4 +296,30 @@ spv_ostream &operator<<(spv_ostream &O, const SPIRVNL &E) {
   return O;
 }
 
+
+// Read the next word from the stream and if OpCode matches the argument,
+// decode the whole instruction. Multiple such instructions are possible.
+// If OpCode doesn't match the argument, set position of the next character to
+// be extracted from the stream to the begining of the non-matching instruction.
```
Comment (svenvh) [inline lib/SPIRV/libSPIRV/SPIRVStream.cpp:303]:
```suggestion
// be extracted from the stream to the beginning of the non-matching instruction.
```

**Review by svenvh — COMMENTED** [review]
There's also a minor style issue that clang-format found.

**Review by AlexeySachkov — APPROVED** [review]

**Review by svenvh — APPROVED** [review]

**Pull request merged**